core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -4203,4 +4204,59 @@ public void testFillUpSequenceIdGapsOnRecovery() throws IOException {
             IOUtils.close(recoveringEngine);
         }
     }
+
+
+    public void assertSameReader(Searcher left, Searcher right) {
+        List<LeafReaderContext> leftLeaves = ElasticsearchDirectoryReader.unwrap(left.getDirectoryReader()).leaves();
+        List<LeafReaderContext> rightLeaves = ElasticsearchDirectoryReader.unwrap(right.getDirectoryReader()).leaves();
+        assertEquals(rightLeaves.size(), leftLeaves.size());
+        for (int i = 0; i < leftLeaves.size(); i++) {
+            assertSame(leftLeaves.get(i).reader(), rightLeaves.get(0).reader());
+        }
+    }
+
+    public void assertNotSameReader(Searcher left, Searcher right) {
+        List<LeafReaderContext> leftLeaves = ElasticsearchDirectoryReader.unwrap(left.getDirectoryReader()).leaves();
+        List<LeafReaderContext> rightLeaves = ElasticsearchDirectoryReader.unwrap(right.getDirectoryReader()).leaves();
+        if (rightLeaves.size() == leftLeaves.size()) {
+            for (int i = 0; i < leftLeaves.size(); i++) {
+                if (leftLeaves.get(i).reader() != rightLeaves.get(0).reader()) {
+                    return; // all is well
+                }
+            }
+            fail("readers are same");
+        }
+    }
+
+    public void testRefreshScopedSearcher() throws IOException {
+        Searcher getSearcher = engine.acquireSearcher("test", Engine.SearcherScope.GET);
+        Searcher searchSearcher = engine.acquireSearcher("test", Engine.SearcherScope.SEARCH);
+        assertSameReader(getSearcher, searchSearcher);
+        IOUtils.close(getSearcher, searchSearcher);
```

Comment:
nit - should we use try-with-resources to avoid leaking on failure?

Reply:
will do